Add more Military Professions, add realistic skills to Military Professions

Beneficial-Insurance opened the pull request:
<!-- HOW TO USE: Under each "#### Heading" below, enter information relevant to your pull request.
Leave the headings unless they don't apply to your PR, and remove the comment blocks (surrounded with <!–– and ––>) when you are done.

NOTE: Please grant permission for repository maintainers to edit your PR.  It is EXTREMELY common for PRs to be held up due to trivial changes being requested and the author being unavailable to make them.
-->

#### Summary
SUMMARY: Content "Military professions"
<!-- This section should consist of exactly one line, formatted like this:

SUMMARY: [Category] "[Briefly describe the change in these quotation marks]"

Do not enter the square brackets [].  Category must be one of these:

- Features
- Content
- Interface
- Mods
- Balance
- Bugfixes
- Performance
- Infrastructure
- Build
- I18N

For more on the meaning of each category, see:
https://github.com/CleverRaven/Cataclysm-DDA/blob/master/doc/CHANGELOG_GUIDELINES.md

If approved and merged, your summary will be added to the project changelog:
https://github.com/CleverRaven/Cataclysm-DDA/blob/master/data/changelog.txt
-->

#### Purpose of change

<!-- With a few sentences, describe your reasons for making this change.  If it relates to an existing issue, you can link it with a # followed by the Github issue number, like #1234.  If your pull request *fully* resolves an issue, include the word "Fix" or "Fixes" before the issue number, like: Fixes #1234.

If there is no related issue, explain here what issue, feature, or other concern you are addressing.  If this is a bugfix, include steps to reproduce the original bug, so your fix can be verified.
-->
The purpose of this PR is to one: add new military professions. Second, to make the starting skillsets of military professions realistic. (A special operator should start with significantly more skill points than six, and skill points in rifles and pistols, dude went through basic training like everyone else and probably used a rifle for a couple of years).
#### Describe the solution

<!-- How does the feature work, or how does this fix a bug?  The easier you make your solution to understand, the faster it can get merged.  -->
All professions that are front line soldiers should start with a three in their weapons skills, in addition to basic first aid and survival skills.
#### Describe alternatives you've considered
<!-- Explain any alternative solutions, different approaches, or possibilities you've considered using to solve the same problem. -->

#### Testing
Loading the edited JSON in a new world checking it doesn't throw errors on load.
<!-- Describe what steps you took to test that this PR resolved the bug or added the feature, and what tests you performed to make sure it didn't cause any regressions.  Also include testing suggestions for reviewers and maintainers.  -->

#### Additional context

anothersimulacrum commented:
> Second, to make the starting skillsets of military professions realistic.

Please note that you are doing this in the title, or do so in another PR (probably preferable).

chaohedgehogs commented on data/json/professions.json:
```diff
@@ -736,23 +763,115 @@
+    "type": "profession",
+    "id": "nco",
+    "name": "Non Comissioned Officer",
+    "description": "You're a veteran of several peace keeping missions.  You lead your squad as a sort of parental figure, offering helpful advice on how not to die.  Now your squad are shambling and want to eat your brains.  ",
```
```suggestion
    "description": "You're a veteran of several peacekeeping missions.  You lead your squad as a sort of parental figure, offering helpful advice on how not to die.  Now your squad are shambling and want to eat your brains.",
```

chaohedgehogs commented on data/json/scenarios.json:
```diff
@@ -528,7 +528,18 @@
+      "mili_pilot",
+      "mili_medic",
+      "mili_burner",
+      "hazmat_unit"
```
```suggestion
      "hazmat_unit"
      "major-general"
      "nco"
```

anothersimulacrum replied:
```suggestion
      "hazmat_unit",
      "major-general",
      "nco"
```

Beneficial-Insurance commented:
@anothersimulacrum, given that in this pull request I replaced the current soldier profession with recruit, and replaced the soldier with a rifleman, the only untouched Military professions is National Guard and Special Ops.... I thought I may as well have a go at giving them relevant skills at the same time?

Beneficial-Insurance commented:
Ok, this PR is looking close to done.
I would like information on how to fix the broken tests, and if my changes look balanced.

ZhilkinSerg commented on data/json/professions.json:
```diff
@@ -696,7 +696,34 @@
-    "id": "soldier",
+    "id": "major-general",
+    "name": "Major General",
+    "description": "You worked your way up through the ranks, from a no-name private, to a big shot Major General.  Now however, it is years since you last fired a weapon in anger, and you barely escaped your aides.  ",
```
```suggestion
    "description": "You worked your way up through the ranks, from a no-name private, to a big shot Major General.  Now however, it is years since you last fired a weapon in anger, and you barely escaped your aides.",
```

ZhilkinSerg commented on data/json/professions.json:
```diff
@@ -706,7 +733,194 @@
+    "type": "profession",
+    "id": "combat-mechanic",
+    "name": "Combat Mechanic",
+    "description": "You failed out of high school, and joined the army.  You were soon hand picked for extra training in the mechanics trade, keeping the armour running. It's been years since you last touched a rifle, and the dead men are marching again...",
```
```suggestion
    "description": "You failed out of high school, and joined the army.  You were soon hand picked for extra training in the mechanics trade, keeping the armour running.  It's been years since you last touched a rifle, and the dead men are marching again...",
```

ZhilkinSerg commented on data/json/professions.json:
```diff
@@ -706,7 +733,194 @@
+    "type": "profession",
+    "id": "combat-engineer",
+    "name": "Combat Engineer",
+    "description": "Your job was simple. Keep the army moving.  You built bridges, you built roads, you destroyed fortifications, and you cleared mines.  It has been years since you last handled a rifle in basic training, now might be the time to dust off those skills.",
```
```suggestion
    "description": "Your job was simple.  Keep the army moving.  You built bridges, you built roads, you destroyed fortifications, and you cleared mines.  It has been years since you last handled a rifle in basic training, now might be the time to dust off those skills.",
```

ZhilkinSerg commented on data/json/professions.json:
```diff
@@ -706,7 +733,194 @@
+    "type": "profession",
+    "id": "nco",
+    "name": "Non Comissioned Officer",
+    "description": "You're a veteran of several peace keeping missions.  You lead your squad as a sort of parental figure, offering helpful advice on how not to die.  Now your squad are shambling and want to eat your brains.  ",
```
```suggestion
    "description": "You're a veteran of several peace keeping missions.  You lead your squad as a sort of parental figure, offering helpful advice on how not to die.  Now your squad are shambling and want to eat your brains.",
```

BrettDong commented:
This is causing some problems with old worlds created before this PR and creating a new custom character in helicopter crash scenario:

<img width="1072" alt="螢幕截圖 2021-01-05 上午6 13 47" src="https://user-images.githubusercontent.com/21075502/103639949-2e0a3980-4f1d-11eb-9ab5-ab2587777246.png">

BrettDong commented:
Probably related to travis CI failure https://travis-ci.org/github/CleverRaven/Cataclysm-DDA/jobs/752978020

ZhilkinSerg commented:
Must be a mod - I do not see `soldier` string anywhere, but in `\data\mods\Dark-Skies-Above\overrides\scenarios.json`.

ZhilkinSerg commented:
I guess we can change id of `rifleman` back to `soldier` (in order not to implement profession migration) or replace `soldier` reference from DSA mod.

BrettDong commented:
I just made #46561 to fix the travis error.
